bundle/config/mutator/python/python_diagnostics.go

```diff
@@ -0,0 +1,129 @@
+package python
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type pythonDiagnostic struct {
+	Severity pythonSeverity `json:"severity"`
+	Summary  string         `json:"summary"`
+	Detail   string         `json:"detail,omitempty"`
+	Location string         `json:"location,omitempty"`
+	Path     string         `json:"path,omitempty"`
+}
+
+type pythonSeverity = string
+
+var locationRegex = regexp.MustCompile(`^(.*):(\d+):(\d+)$`)
+
+const (
+	pythonError   pythonSeverity = "error"
+	pythonWarning pythonSeverity = "warning"
+)
+
+// parsePythonDiagnostics parses diagnostics from the Python mutator.
+//
+// diagnostics file is newline-separated JSON objects with pythonDiagnostic structure.
+func parsePythonDiagnostics(input io.Reader) (diag.Diagnostics, error) {
+	// the default limit is 64 Kb which should be enough for diagnostics
+	scanner := bufio.NewScanner(input)
+	diagnostics := diag.Diagnostics{}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line == "" {
+			continue
+		}
+
+		parsedLine := pythonDiagnostic{}
+
+		err := json.Unmarshal([]byte(line), &parsedLine)
```

Comment:
You can use https://pkg.go.dev/encoding/json#NewDecoder and not deal with the newlines.

It works with line-delimited JSON objects out of the box.

Reply:
Thanks! This is much easier now